src/spring-cloud/azext_spring_cloud/custom.py

```diff
@@ -1596,83 +1629,72 @@ def domain_unbind(cmd, client, resource_group, service, app, domain_name):
-            trace_enabled=True, app_insights_instrumentation_key=app_insights_key)
+        monitoring_setting_properties = models_20201101preview.MonitoringSettingProperties(
+            trace_enabled=True,
+            app_insights_instrumentation_key=app_insights_key)
     elif app_insights:
-        if is_valid_resource_id(app_insights):
-            resource_id_dict = parse_resource_id(app_insights)
-            instrumentation_key = get_app_insights_key(cmd.cli_ctx, resource_id_dict['resource_group'],
-                                                       resource_id_dict['resource_name'])
-            trace_properties = models_20201101preview.MonitoringSettingProperties(
-                trace_enabled=True, app_insights_instrumentation_key=instrumentation_key)
-        else:
-            instrumentation_key = get_app_insights_key(cmd.cli_ctx, resource_group, app_insights)
-            trace_properties = models_20201101preview.MonitoringSettingProperties(
-                trace_enabled=True, app_insights_instrumentation_key=instrumentation_key)
-    elif enable_java_agent is True:
-        create_app_insights = True
-
-    if create_app_insights is True:
-        try:
-            instrumentation_key = try_create_application_insights(cmd, resource_group, service_name, location)
-            if instrumentation_key:
-                trace_properties = models_20201101preview.MonitoringSettingProperties(
-                    trace_enabled=True, app_insights_instrumentation_key=instrumentation_key)
-        except Exception:  # pylint: disable=broad-except
-            logger.warning(
-                'Error while trying to create and configure an Application Insights for the Azure Spring Cloud. '
-                'Please use the Azure Portal to create and configure the Application Insights, if needed.')
-            return None
-    return trace_properties
+        connection_string = _get_connection_string_from_app_insights(cmd, resource_group, app_insights)
+        monitoring_setting_properties = models_20201101preview.MonitoringSettingProperties(
+            trace_enabled=True,
+            app_insights_instrumentation_key=connection_string)
+    return monitoring_setting_properties
+
+
+def _get_connection_string_from_app_insights(cmd, resource_group, app_insights):
+    """Get connection string from:
+    1) application insights name
+    2) application insights resource id
+    """
+    if is_valid_resource_id(app_insights):
+        resource_id_dict = parse_resource_id(app_insights)
+        connection_string = get_app_insights_connection_string(
+            cmd.cli_ctx, resource_id_dict['resource_group'], resource_id_dict['resource_name'])
+    else:
+        connection_string = get_app_insights_connection_string(cmd.cli_ctx, resource_group, app_insights)
+    if not connection_string:
+        raise CLIError("Cannot find Connection string from application insights:{}".format(app_insights))
```

Comment:
Please use a specific error type instead of `CLIError`

Reply:
Updated

Reply:
Updated